src/powershell/Public/Remove-FinOpsCostExport.ps1

```diff
@@ -0,0 +1,169 @@
+      -ResourceType "exports" `
+      -Name $Name `
+      -ApiVersion $ApiVersion `
+      -Method "GET" `
+      -Payload $payload
+
+    if ($httpResponse.StatusCode -eq 200)
+    {
+      # Export deleted successfully
+    }
+    elseif ($httpResponse.StatusCode -eq 404)
+    {
+      # Not found
+    }
+    else
+    {
+      # Error response describing why the operation failed.
+      throw "Not Exists: Cost Management Export operation failed with message: '$($httpResponse.Content)'"
+    }
+
+    if ($PSCmdlet.ShouldProcess($Name, 'DeleteCostExport'))
+    {
+
+      # Using the REST API to delete the export as requested as PS modules are outdated?
+      $httpResponse = Invoke-AzRestMethod `
+        -ResourceProviderName "Microsoft.CostManagement" `
+        -ResourceType "exports" `
+        -Name $Name `
+        -ApiVersion $ApiVersion `
+        -Method "DELETE" `
+        -Payload $payload
+
+      if ($httpResponse.StatusCode -eq 200)
+      {
+        # Export deleted successfully
+      }
+      elseif ($httpResponse.StatusCode -eq 404)
+      {
+        # Not found - nothing to delete
+      }
+      else
+      {
+        # Error response describing why the operation failed.
+        throw "Delete Cost Management Export operation failed with message: '$($httpResponse.Content)'"
+      }
+    }
+
+    # Delete associated ingestion data from storage account
+    if ($RemoveData)
+    {
+      # Using the REST API to get the export as requested as PS modules are outdated?
+      $httpResponse = Invoke-AzRestMethod `
+        -ResourceProviderName "Microsoft.CostManagement" `
+        -ResourceType "exports" `
+        -Name $Name `
+        -ApiVersion $ApiVersion `
+        -Method "GET" `
+        -Payload $payload
+
+      if ($httpResponse.StatusCode -eq 200)
+      {
+
+        # Export details retreived
+        $exportDetails = ConvertFrom-Json -InputObject $httpResponse.Content
+        $storageAccountID = $exportDetails.properties.deliveryInfo.destination.resourceId
+
+        # Get-AzStorageAccount -resourcegroupname alz-finopstk -name finopshub255k7ov6asado | Get-AzDataLakeGen2ChildItem -FileSystem "ingestion" -Path "/" -Recurse -FetchProperty | Where-Object IsDirectory -ne false | Select-Object Path
+        $resourceGroupName = $storageAccountID.Split('/')[4]
+        $storageAccountName = $storageAccountID.Split('/')[8]
+
+        #Hold on to your hats, deleting all the files in the ingestion scope. Using Az PS module as this should be current.
+        if ($PSCmdlet.ShouldProcess($scope, 'DeleteCostReports'))
+        {
+          Write-Verbose "Resourace Group: $resourceGroupName"
+          Write-Verbose "Storage Account: $storageAccountName"
+          Write-Verbose "Scope: $scope"
+
+          $getFiles = Get-AzStorageAccount -ResourceGroupName $resourceGroupName -Name $storageAccountName | Get-AzDataLakeGen2ChildItem -FileSystem "ingestion" -Path $scope -Recurse -FetchProperty
```

Comment:
Is the "ingestion" file system an ADLS thing or a hubs thing? Just want to make sure we're not tying this generic exports command to hubs. Also note that not all storage accounts are ADLS, so we might need to check to see if it is ADLS first.

Reply:
I was missing this context. Yes, File System (container) is required when interacting with ADLS storage accounts.